src/owl2Shacl.py

```diff
@@ -0,0 +1,224 @@
+    owl_classes = ont_graph.subjects(predicate=RDF.type, object=OWL.Class)
+    classes = chain(rdf_classes, owl_classes)
+
+    for item in classes:
+        if type(item) is not BNode:  # only create shapes for identified classes
+            sh_graph.add((URIRef(str(item) + 'Shape'), RDF.type, SH.NodeShape))
+            sh_graph.add((URIRef(str(item) + 'Shape'), SH.targetClass, item))
+            sh_graph.add((URIRef(str(item) + 'Shape'), SH.severity, SH.Warning))
+
+            superclasses = ont_graph.objects(subject=item, predicate=RDFS.subClassOf)
+            for superclass in superclasses:
+                if type(superclass) is BNode:  # i.e. it is a restriction
+                    restriction_details = bind_restriction_values(ont_graph, superclass)
+                    add_restriction(ont_graph, sh_graph, item, restriction_details)
+
+
+def create_shacl(ontology: str | Path | Graph) -> Tuple[Graph, Graph]:
+    if isinstance(ontology, (Path, str)):
+        ont_graph = Graph().parse(ontology)
+    else:
+        ont_graph = ontology
+    sh_graph = Graph()
+    # bind namespaces from ontology to shape graph
+    for name, ns in ont_graph.namespaces():
+        sh_graph.bind(name, ns, replace=True)
+
+    _create_node_shapes_for_classes(ont_graph, sh_graph)
+    _create_node_shapes_for_properties(ont_graph, sh_graph, OWL.ObjectProperty)
+    _create_node_shapes_for_properties(ont_graph, sh_graph, OWL.DatatypeProperty)
+
+    return ont_graph, sh_graph
+
+
+def rdf_validate(data_file: str | Graph, ont_graph: str | Graph, sh_graph: str | Graph) -> Tuple[
+        bool, Graph, str]:
+    # run shacl validation
+    conforms, results_graph, results_text = validate(data_file,
+                                                     shacl_graph=sh_graph,
+                                                     ont_graph=ont_graph,
+                                                     inference='none',
+                                                     abort_on_first=False,
+                                                     allow_infos=False,
+                                                     allow_warnings=False,
+                                                     meta_shacl=False,
+                                                     advanced=True,
+                                                     js=False,
+                                                     debug=False)
+
+    return conforms, results_graph, results_text
+
+
+def _parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-o", "--ontology", help="Path to ontology - required", required=True)
+    parser.add_argument("-v", "--validate", help="Path of file to validate", default=None)
+    parser.add_argument("-s", "--shacl", help="Location for resulting shapes graph", default=None)
+
+    return parser.parse_args()
+
+
+def main(argv):
+    args = _parse_args()
+    if args.ontology:
+        ont_graph, sh_graph = create_shacl(args.ontology)
```

Comment:
suggest you add a `--dryrun` arg that outputs useful info on the shacl shapes generated (maybe just counts for now?) and doesn't write the file.